test flake runner test flake

Seen in a CI run:

https://github.com/tailscale/tailscale/actions/runs/5403629678/jobs/9816703048?pr=8473
```
=== RUN   TestFlakeRun
    flakytest.go:38: flakytest: issue tracking this flaky test: https://github.com/tailscale/tailscale/issues/0
    flakytest_test.go:38: failing on purpose
--- FAIL: TestFlakeRun (0.00s)
=== RUN   TestLoopbackSOCKS5
    tsnet_test.go:107: DERP httpsrv listener: 127.0.0.1:45783
    tsnet_test.go:119: testcontrol listening on http://127.0.0.1:[45](https://github.com/tailscale/tailscale/actions/runs/5403629678/jobs/9816703048?pr=8473#step:8:46)[46](https://github.com/tailscale/tailscale/actions/runs/5403629678/jobs/9816703048?pr=8473#step:8:47)1
    tsnet_test.go:385: socks connect tcp 127.0.0.1:39125->100.64.0.1:8081: unknown error general SOCKS server failure
    stuntest.go:63: STUN server shutdown
--- FAIL: TestLoopbackSOCKS5 (5.55s)
```

wut? 😂 

cc @maisem

--- a/cmd/testwrapper/flakytest/flakytest_test.go
+++ b/cmd/testwrapper/flakytest/flakytest_test.go
@@ -28,6 +28,9 @@ func TestIssueFormat(t *testing.T) {
 	}
 }
 
+// TestFlakeRun is a test that fails when run in the testwrapper
+// for the first time, but succeeds on the second run.
+// It's used to test whether the testwrapper retries flaky tests.
 func TestFlakeRun(t *testing.T) {
 	Mark(t, "https://github.com/tailscale/tailscale/issues/0") // random issue
 	e := os.Getenv(FlakeAttemptEnv)
@@ -35,6 +38,6 @@ func TestFlakeRun(t *testing.T) {
 		t.Skip("not running in testwrapper")
 	}
 	if e == "1" {
-		t.Fatal("failing on purpose")
+		t.Fatal("First run in testwrapper, failing so that test is retried. This is expected.")
 	}
 }

